### DISCUSSION TITLE
cluster_table.data配置子集群时，Addr可否为空

### DISCUSSION THREAD

**Discussion by haoxuebing** [bfenetworks/bfe#870] [Q&A (问题和回答)] [answered]
我有个外网域名 a.com  和 内网域名 b.com，外网通过a.com访问b.com，b.com的服务下有自己的负载均衡

cluster_table.data配置子集群中的Addr填写b.com服务的容器ip服务正常，但Addr不填或者填写b.com的网关ip就会报 400错误（Request Header Or Cookie Too Large）

请问这是什么原因造成的？

**Comment by Tovi163** [comment, marked as answer]
1. Addr 不填写，将找不到后端的RS
2. 容器IP和网关IP的不同，应该是400的一个因素，可以想办法排除下

**Reply by haoxuebing** [reply]
子集群一定要填写容器IP吗，可是容器IP是不固定的，每次手动填太麻烦了吧

是不是有其他方式实现呢？

**Reply by mileszhang2016** [reply]
建议和已经开源的BFE控制面打通，通过调用[BFE API-Server](https://github.com/bfenetworks/api-server)的接口来更新RS的地址。
API接口的定义我们会尽快发布出来。

**Reply by haoxuebing** [reply]
好的，我试下

**Reply by daimg** [reply]
> 子集群一定要填写容器IP吗，可是容器IP是不固定的，每次手动填太麻烦了吧
> 
> 是不是有其他方式实现呢？


如果从bfe到b.com的网关的网络是可达的，bfe应该是可以使用网关的ip连接后端服务容器的。建议你可以也检查下网关，比如日志，看看网关是什么原因拒绝了从bfe来的请求消息。

**Reply by haoxuebing** [reply]
之前用的是七层协议的负载均衡，内网网关nginx配置问题导致的400，换成四层协议的负载均衡，这个问题就解决了，感谢各位大佬
